frontend/src/pages/Main.tsx

```diff
@@ -22,13 +28,44 @@ const Main: FC = () => {
   const [moveToPosition, setMoveToPosition] = useState<LatLng>();
   // The indicator to display on the map
   const [mode, setMode] = useState<string>('ALL');
+  // The indicator to display on the map
+  const [multiMode, setMultiMode] = useState<MultiMode>();
   // The selected range of date (used to filter the data to show)
   const [rangeSelected, setRangeSelected] = useState<DateRange>({});
 
+  function multiModeSet(selected: any) {
+    const selectedOptions: any = selected;
+
+    const outputMode: MultiMode = {
+      count: selectedOptions.length,
+      mode: selectedOptions.map((e: any) => e.value),
+      ALL: selectedOptions.some((e: any) => e.value === 'ALL'),
+      KPI: selectedOptions.some((e: any) => e.value === 'KPI'),
+      DI: selectedOptions.some((e: any) => e.value === 'DI'),
+      IRI: selectedOptions.some((e: any) => e.value === 'IRI'),
+      Mu: selectedOptions.some((e: any) => e.value === 'Mu'),
+      Enrg: selectedOptions.some((e: any) => e.value === 'E_norm'),
+    };
+
+    // handles single condition indicator or ALL
+
+    if (outputMode.ALL === true) {
+      setMode('ALL');
+    } else if (outputMode.count === 0) {
+      setMode(' ');
+    } else if (outputMode.ALL === false && outputMode.count === 1) {
+      const mode: string = selectedOptions.map((e: any) => e.value).toString();
+      setMode(mode);
+    } else {
+      setMultiMode(outputMode);
+    }
+    console.log(multiMode);
+  }
+
   function dateChange(date: any) {
     const YearMonth: YearMonth = {
       year: date.getFullYear(),
-      month: date.getMonth() + 1, // +1 why
+      month: date.getMonth() + 1,
```

Comment:
do the months start at 0? so you have to add 1?

Reply:
Yes! https://stackoverflow.com/questions/2552483/why-does-the-month-argument-range-from-0-to-11-in-javascripts-date-constructor